Formula/pipenv.rb

```diff
@@ -98,6 +98,7 @@ def post_install
   test do
     ENV["LC_ALL"] = "en_US.UTF-8"
     assert_match "Commands", shell_output("#{bin}/pipenv")
+    system "#{bin}/pipenv", "--python", Formula["python@3.10"].opt_bin/"python3"
```

Comment:
This is a bit concerning. This makes me suspect Python3.10 isn't being used at runtime on macOS either (not without further configuration, at least).

Reply:
The issue is not on macOS os but it is on Linux (testing) where the default python is the 3.5 (old Ubuntu 16.04).
And pipenv uses the default python when not expecified
And this becomes a issue now because pipenv only supports python >= 3.6

Reply:
But which Python is it using on macOS?

Reply:
for sure its not the python 3.10 because its formula is keg-only.
checking the build... it was installing python 3.8, 3.9 and 3.10.
I am quite confident it was using the 3.9

<img width="1110" alt="Screenshot 2021-11-10 at 03 05 25" src="https://user-images.githubusercontent.com/1225902/141036807-34a9be3b-f018-4114-bae1-963c7b0379c0.png">

Reply:
I am basing myself on this docs

> If you don't specify a Python version on the command–line, either the [requires] python_full_version or python_version will be selected automatically, falling back to whatever your system's default python installation is, at time of execution.

https://github.com/pypa/pipenv/blob/main/docs/basics.rst#-specifying-versions-of-python

Reply:
`python@3.8` and `python@3.9` are never actually installed though. That's why there's a `--force` in `brew uninstall` to avoid erroring out when you try to uninstall something that isn't installed. You can try the `brew install` steps shown to see.

Reply:
to clarify, there are two different `python`'s at play here:

1. the `python` that `pipenv` itself runs under (constant, brew's `python@3.10`)
1. the `python` that `pipenv` selects for a new project, creating/managing a virtualenv (variable, dependent on available interpreters)

what we're seeing is that (1) appears to be changed base on (2), which is definitely unexpected. we need to figure out why

Reply:
Thanks @branchvincent for clarify

But (1) is not been changed. It's been always 3.10

(2) when no --python argument it gets the default from system, which might be even apple pre-installed python on macOS and on case of Linux the 3.5

If python 3.10 was not keg-only it would use it.

We can test it changing the depends on to python 3.9 and remove the --python argument

Reply:
i was able to reproduce this with a non-homebrew install and created https://github.com/pypa/pipenv/issues/4854. @bertonha, feel free to check out that issue and verify that you can also reproduce the same issue

Reply:
@branchvincent do you still see this as an issue to not merge ?

Reply:
that issue was closed, apparently this is expected behavior 🤷‍♂️